components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -7111,6 +7112,58 @@ class _ImageWrapper (BlitzObjectWrapper, OmeroRestrictionWrapper):
     PLANEDEF = omero.romio.XY
 
     @classmethod
+    def _getQueryString(cls, opts=None):
+        """
+        Extend base query to handle filtering of Images by Datasets.
+        Returns a tuple of (query, clauses, params).
+        Supported opts: 'dataset': <dataset_id> to filter by Dataset
+                        'load_pixels': <bool> to load Pixel objects.
+                        'load_channels': <bool> to load Channels and
+                                                    Logical Channels
+                        'orphaned': <bool>. Images not in Dataset or WellSample
+
+        :param opts:        Dictionary of optional parameters.
+        :return:            Tuple of string, list, ParametersI
+        """
+        query, clauses, params = super(
+            _ImageWrapper, cls)._getQueryString(opts)
+        if opts is not None and 'dataset' in opts:
```

Comment:
In this case ```opts['dataset']``` dataset is not a boolean so I don't want to check the truthiness of ```opts.get('dataset')```, but just check whether the value is set. E.g. if ```opts.get('dataset')``` is 0 this would be evaluated as ```False```, even though I'd want to use that value.
Unless we decide that we consider ```0``` to be invalid, but that is a change in functionality (not just a stylistic change)?